test/checkout.int.t.sol

```diff
@@ -1119,8 +1171,150 @@ contract Checkout_swapFromSupplierWithoutFeeSpecialOrder is Checkout {
     );
     assertEq(_bpNori.balanceOf(_namedAccounts.feeWallet), 0);
   }
+
+  function test_revertsWhenSupplierDoesNotExistInMarket() external {
+    vm.prank(owner);
+    vm.expectRevert(InsufficientSupply.selector);
+    _market.swapWithoutFeeSpecialOrder({
+      recipient: owner,
+      purchaser: owner,
+      amount: certificateAmount,
+      customFee: customFee,
+      customPriceMultiple: customPriceMultiple,
+      supplier: _namedAccounts.supplier2, // not listed in market
+      vintages: new uint256[](0)
+    });
+  }
+}
+
+contract Checkout_swapWithoutFeeSpecialOrder_specificVintagesSpecificSupplier is
+  Checkout
+{
+  uint256 ownerPrivateKey = 0xA11CE;
+  address owner = vm.addr(ownerPrivateKey);
+  uint256 customFee = 5;
+  uint256 certificateAmount = 1.5 ether;
+  uint256[] vintages = [2019, 2020];
+  uint256 _priceMultiple = 2000;
+
+  function setUp() external {
+    /**
+     * Supplier 1: 2018 vintage
+     * Supplier 2: 2018, 2019 and 2020 vintage
+     * Supplier 3: 2019 vintage
+     * All removals are 1 tonne
+     * Certificate amount is 1.5 tonnes
+     * Fulfillment should result in all of supplier2's 2019 removal being used, and .5 tonne of supplier2's 2020 removal
+     */
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier,
+        amount: 1 ether,
+        vintage: 2018
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier2,
+        amount: 1 ether,
+        vintage: 2018
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier2,
+        amount: 1 ether,
+        vintage: 2019
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier2,
+        amount: 1 ether,
+        vintage: 2020
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier3,
+        amount: 1 ether,
+        vintage: 2019
+      })
+    );
+    uint256 purchaseAmount = _market.calculateCheckoutTotalWithoutFee({
+      amount: certificateAmount,
+      priceMultiple: _priceMultiple
+    });
+    _market.grantRole({role: _market.MARKET_ADMIN_ROLE(), account: owner});
+    vm.prank(_namedAccounts.admin);
+    _bpNori.deposit(owner, abi.encode(purchaseAmount));
+    vm.prank(owner);
+    _bpNori.approve(address(_market), purchaseAmount);
+  }
+
+  function test_basicFulfillment() external {
+    vm.prank(owner);
+    vm.recordLogs();
+    _market.swapWithoutFeeSpecialOrder(
+      owner,
+      owner,
+      certificateAmount,
+      customFee,
+      _priceMultiple,
+      _namedAccounts.supplier2,
+      vintages
+    );
+
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    // In this test scenario, the `CreateCertificate` event index is 18 because multiple suppliers are getting
```

Comment:
I think we should just use 18 statically instead of looping. If the test breaks because we added more events or changed logic otherwise, it's probably a good thing to be alerted about. 

Worried about this test not catching multiple certificate creation events, using 0 as the index,  etc otherwise. 

Thoughts?